src/app/modules/spcp/sp.oidc.service.ts

```diff
@@ -0,0 +1,310 @@
+import fs from 'fs'
+import { err, errAsync, ok, okAsync, Result, ResultAsync } from 'neverthrow'
+
+import {
+  ISpcpMyInfo,
+  spcpMyInfoConfig,
+} from '../../config/features/spcp-myinfo.config'
+import { createLoggerWithLabel } from '../../config/logger'
+
+import { SpOidcClient } from './sp.oidc.client'
+import { PublicJwks } from './sp.oidc.client.types'
+import {
+  CreateJwtError,
+  CreateRedirectUrlError,
+  ExchangeAuthTokenError,
+  InvalidIdTokenError,
+  InvalidJwtError,
+  InvalidStateError,
+  MissingJwtError,
+  VerifyJwtError,
+} from './spcp.errors'
+import {
+  JwtName,
+  JwtPayload,
+  JwtPayloadFromCookie,
+  ParsedSpcpParams,
+  SingpassJwtPayloadFromCookie,
+  SpcpCookies,
+  SpcpDomainSettings,
+} from './spcp.types'
+import { extractFormId, isSingpassJwtPayload } from './spcp.util'
+
+const logger = createLoggerWithLabel(module)
+
+/**
+ * Class for executing Singpass OIDC-related services.
+ * Exported for testing.
+ */
+export class SpOidcServiceClass {
+  #spOidcProps: ISpcpMyInfo
+  #spOidcClient: SpOidcClient
+
+  constructor(props: ISpcpMyInfo) {
+    this.#spOidcClient = new SpOidcClient({
+      spOidcRpClientId: props.spOidcRpClientId,
+      spOidcRpRedirectUrl: props.spOidcRpRedirectUrl,
+      spOidcNdiDiscoveryEndpoint: props.spOidcNdiDiscoveryEndpoint,
+      spOidcNdiJwksEndpoint: props.spOidcNdiJwksEndpoint,
+      spOidcRpPublicJwks: JSON.parse(
+        fs.readFileSync(props.spOidcRpJwksPublicPath).toString(),
+      ),
+      spOidcRpSecretJwks: JSON.parse(
+        fs.readFileSync(props.spOidcRpJwksSecretPath).toString(),
+      ),
+    })
+    this.#spOidcProps = props
```

Comment:
nit: i think there is no need to pass in all the props right? i feel like we should only give the props the class needs

from a quick scan, only 3 subprops are used: 
`spCookieMaxAgePreserved`, `spCookieMaxAge` and `spcpCookieDomain`